ledger/src/bank_forks.rs

```diff
@@ -219,6 +234,15 @@ impl BankForks {
         );
     }
 
+    fn check_interval(root: Slot, parent_slot: Slot, snapshot_interval_slots: usize) -> bool {
+        let interval = snapshot_interval_slots as u64;
+        assert!(parent_slot < root || root == 0);
+        assert!(interval > 0);
```

Comment:
IIRC, `config.snapshot_interval_slots = 0` is valid value meaning to disable snapshots altogether. It is invalid now?

Reply:
Yes we still overload 0 to mean, disable snapshots

Reply:
snapshot_config should be translated to None if snapshot_interval_slots=0, so I don't think this function needs to handle the interval=0 case.

Reply:
@sakridge Sadly, it was not translated to `None` but the current code effectively disables the condition so I think it's ok:

```
[2020-03-02T07:00:27.184816585Z ERROR solana_ledger::bank_forks] snapshot_config: Some(SnapshotConfig { snapshot_interval_slots: 18446744073709551615, snapshot_package_output_path: "/home/ryoqun/work/solana/solana/config/ledger", snapshot_path: "/home/ryoqun/work/solana/solana/config/ledger/snapshot" })
```